com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaBuilder.java

```diff
@@ -221,14 +221,11 @@ public void build(Task<Void> task) throws CompileExceptionError, IOException {
         builder.setProperties(propertiesMsg);
         builder.addAllPropertyResources(propertyResources);
 
-        // create any builder plugins
-        if (luaBuilderPlugins == null) {
-            luaBuilderPlugins = PluginScanner.createPlugins("com.dynamo.bob.pipeline", LuaBuilderPlugin.class);
-        }
-        // apply any builder plugins
-        for(LuaBuilderPlugin builderPlugin : luaBuilderPlugins) {
+        // create and apply a builder plugin if one exists
+        LuaBuilderPlugin luaBuilderPlugin = PluginScanner.createPlugin("com.dynamo.bob.pipeline", LuaBuilderPlugin.class);
```

Comment:
Why recreate it for every script?
You have the static variable, I assume you intended to use that one?

Reply:
Edit: I now see that you've created a cache inside the create function.
I think the name should be then called `createOrGet()` to communicate that.